tests/forms_tests/tests/test_forms.py

```diff
@@ -1904,6 +1904,28 @@ class UserRegistration(Form):
 </select></li>"""
         )
 
+    def test_callable_initial_data_bound_form_disabled_field(self):
+        def initial_name():
+            return 'John Doe'
+
+        class PersonForm(Form):
+            name = CharField(initial=initial_name, disabled=True)
+
+        # Assert base case without form data is correct.
+        form = PersonForm(auto_id=False)
+        self.assertHTMLEqual(
+            form.as_ul(),
+            '<li>Name: <input name="name" type="text" value="John Doe" disabled required /></li>',
```

Comment:
Could you try to write these tests at a lower level, without the `as_ul()` output methods? They're difficult to debug now.